conjure-java-core/src/integrationInput/java/test/prefix/com/palantir/product/DoubleAliasExample.java

```diff
@@ -26,7 +26,9 @@ public String toString() {
     @Override
     public boolean equals(Object other) {
         return this == other
-                || (other instanceof DoubleAliasExample && this.value == ((DoubleAliasExample) other).value);
+                || (other instanceof DoubleAliasExample
+                        && Double.doubleToLongBits(this.value)
+                                == Double.doubleToLongBits(((DoubleAliasExample) other).value));
```

Comment:
This is what the boxed `Double.equals` does under the hood:

http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Double.java#l797

Reply:
Could we use `Double.compare()`?

Reply:
We could, but not sure what the benefit would be since it's just using this `doubleToLongBits` method under the hood anyway?

```java
    public static int compare(double d1, double d2) {
        if (d1 < d2)
            return -1;           // Neither val is NaN, thisVal is smaller
        if (d1 > d2)
            return 1;            // Neither val is NaN, thisVal is larger

        // Cannot use doubleToRawLongBits because of possibility of NaNs.
        long thisBits    = Double.doubleToLongBits(d1);
        long anotherBits = Double.doubleToLongBits(d2);

        return (thisBits == anotherBits ?  0 : // Values are equal
                (thisBits < anotherBits ? -1 : // (-0.0, 0.0) or (!NaN, NaN)
                 1));                          // (0.0, -0.0) or (NaN, !NaN)
    }
```

Reply:
you get to avoid the more expensive `Double.doubleToLongBits` on the common path where values are different and we get to simplify generated code

Reply:
@ferozco We could do some benchmarking to figure out whether the branching of checking d1 < d2 is faster than just running the `doubleToLongBits` every time, but given that in the jdk's Double.equals method they decided to use:
```
    public boolean equals(Object obj) {
        return (obj instanceof Double)
               && (doubleToLongBits(((Double)obj).value) ==
                      doubleToLongBits(value));
    }
```

I'm kind inclined to just do the same thing they picked?

Reply:
Performance should be equivalent outside of tight loops -- cost of doubleToRawLongBits is near zero.